pkg/ipamd/ipamd.go

```diff
@@ -1332,15 +1342,27 @@ func (c *IPAMContext) SetNodeLabel(key, value string) error {
 	}
 
 	// Update node status to advertise the resource.
-	_, err = c.k8sClient.CoreV1().Nodes().Update(updateNode)
+	err = c.k8sClient.Update(ctx, updateNode)
 	if err != nil {
 		log.Errorf("Failed to update node %s with label %q: %q, error: %v", c.myNodeName, key, value, err)
 	}
-	log.Infof("Updated node %s with label %q: %q", c.myNodeName, key, value)
+	log.Debugf("Updated node %s with label %q: %q", c.myNodeName, key, value)
+
 	return nil
 }
 
 // GetPod returns the pod matching the name and namespace
-func (c *IPAMContext) GetPod(podName, namespace string) (*v1.Pod, error) {
-	return c.k8sClient.CoreV1().Pods(namespace).Get(podName, metav1.GetOptions{})
+func (c *IPAMContext) GetPod(podName, namespace string) (*corev1.Pod, error) {
+	ctx := context.Background()
```

Comment:
This is _definitely_ not an async background operation :stuck_out_tongue: 
Make this `context.TODO()` to indicate that we need to update the code to take ctx as an argument (at some point).